core/isolate.rs

```diff
@@ -440,10 +440,7 @@ impl Isolate {
       libdeno::deno_mod_new(self.libdeno_isolate, main, name_ptr, source_ptr)
     };
 
-    self.check_last_exception().map(|_| id).map_err(|err| {
-      assert_eq!(id, 0);
-      err
-    })
+    self.check_last_exception().map(|_| id)
```

Comment:
Since this is a bug in core, it should be testable with a unit test in core.

Reply:
I extended existing `dyn_import_err` unit test to demonstrate that you can do multiple dynamic imports that end with error.